firmware/application/apps/ui_recon.cpp

```diff
@@ -40,8 +40,8 @@ void ReconView::clear_freqlist_for_ui_action() {
     audio::output::stop();
     // flag to detect and reload frequency_list
     if (!manual_mode) {
-        frequency_list.clear();
-        frequency_list.shrink_to_fit();
+        // clear and shrink_to_fit are not enough to really start with a new, clean, empty vector
+        // swap is the only way to achieve a perfect memory liberation
```

Comment:
Just a thought here, if you're going to be re-filling this and it's being passed by reference now, you might as well just call clear and then let it get filled then shrink to fit. Otherwise you have to realloc again. It's minor, but you can possibly avoid some unnecessary reallocation now.

Reply:
Let met try :-)

Reply:
I'm wondering... if you could first go through the file once and count the lines, then swap to empty and reserve the right size, that would be the less allocations, and less fragmentation... but on the other hand you have to read the file twice which is probably still slower than reallocating and copying the vector contents a couple of times

Reply:
Not better :-/